### DISCUSSION TITLE
422 Unprocessable Entity with Proper Input

### DISCUSSION THREAD

**Discussion by aminPial** [fastapi/fastapi#7434] [Questions] [answered]
Thanks for this super cool Framework. Anyway, I am getting really frustrated on FastAPI bugs as we are getting these bugs in our production server!

An Api is getting called from android devices. Following is simple Plain Old Java Class to define the JSON Post Request Body:

```
    @SerializedName("longitude")
    double longitude;

    @SerializedName("latitude")
    double latitude;

    @SerializedName("radius")
    double radius;

    @SerializedName("page_no")
    int page_no;

    @SerializedName("search_query")
    String search_query;

    @SerializedName("user_token")
    String user_token;
```

And my Pydantic Model is defined like this:

```
class TextSearchQuery(BaseModel):

    longitude: float

    latitude: float

    radius: float

    page_no: int

    search_query: str

    user_token: str
```
What is Wrong? I don't really get this! Really frustrated with this bug again and again!

`INFO:     37.111.237.7:0 - "POST /api/v2/text_search HTTP/1.1" 422 Unprocessable Entity`

 If i put **Type Any** for all then this works! but then the whole point of using pydantic model is meaningless!

 **Does FastAPI has some easy way to just get the JSON POST request body like flask?. Using Pydantic Model is giving us too much hassle and bugs. And causing problems for around 50K+ Users and it will be meaningless to just migrate our codebase to flask just for some couple of bugs!**  Anyway, Thanks again for this cool Project!

**Comment by tiangolo** [comment, marked as answer]
Well, it seems your clients are sending invalid data. Check the responses, they will contain where's the invalid data exactly.

Anyway, I assume you solved your problem, so thanks for closing the issue.
